src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -1127,23 +1127,22 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
         distinct = true;
       }
     }
-    ExpressionPtr result;
+    auto makePtr = [&]<typename ExpressionType>(auto&&... additionalArgs) {
+      ExpressionPtr result{std::make_unique<ExpressionType>(
+          distinct, std::move(childExpression), AD_FWD(additionalArgs)...)};
+      result->descriptor() = context->getText();
+      return result;
+    };
     if (ad_utility::getLowercase(children[0]->getText()) == "count") {
-      result =
-          ExpressionPtr{std::make_unique<sparqlExpression::CountExpression>(
-              distinct, std::move(childExpression))};
+      return makePtr.operator()<sparqlExpression::CountExpression>();
```

Comment:
I've found an even better way:
Give the `makePtr` a first (unnamed and unused) Parameter of type `ExpressionType*` (before the additionalArgs), 
then your call becomes:
`return makePtr(CountExpression*{});` which is arguably simpler.
And then (again, not your fault, but your chance to make things better),
assign the result of `getLowercase(children[0]->getText())` to a variable and reuse it in the if-else-checks.

As you can see I've grown to dislike code duplication for various reasons, and there is a lot of room for improvements in the QLever codebase.